### PR TITLE
feat(mlx): add OPENMED_MLX_MMAP toggle for memory-mapped weight loading

### LINKED ISSUE #296: Add memory-mapped weight loading for MLX artifacts
## Summary
On phone/laptop tiers (section 6.2) cold-start and peak RAM are dominated by how weights are read. openmed/mlx/models/__init__.load_model currently does mx.load(str(weights_path)) and then load_weights, materializing the full tensor set up front. For the Tiny/Nano tiers a memory-mapped safetensors read reduces peak resident set and speeds first-load. This is a focused, dependency-light hardening of the existing MLX loader.

## Scope
- [ ] Switch the safetensors weight read in load_model to a memory-mapped path where MLX supports it (mx.load with mmap, or lazy load), with a graceful fallback to the current eager path.
- [ ] Ensure quantized-checkpoint detection (_is_quantized_checkpoint) and load still work under the mmap path.
- [ ] Add a config/env toggle (OPENMED_MLX_MMAP) so callers can force eager loading for debugging.
- [ ] Add a test asserting weights still load correctly (same logits on a stub) with mmap enabled and disabled.

## Acceptance criteria
- [ ] load_model uses a memory-mapped read for safetensors with a documented fallback.
- [ ] Quantized and non-quantized stubs both load correctly under mmap.
- [ ] OPENMED_MLX_MMAP=0 forces the eager path.
- [ ] test suite green: .venv/bin/python -m pytest tests/ -q

## Out of scope
- CoreML/ONNX loaders (separate).
- Changing the artifact on-disk format.
- Benchmarking the RAM win (covered by the device-bench / cold-start tasks).

## Files
- openmed/mlx/models/__init__.py
- tests/unit/mlx/test_mmap_loading.py

---
Task: OM-131  ·  Milestone: Backlog  ·  Priority: P2  ·  Size: S
Depends on: —  ·  Blocks: —
Roadmap: section 6.2 (RAM/cold-start budgets)
Spec: PLANS/V2/EXECUTION/tasks/OM-131.md

### PR DESCRIPTION
**Pull request by DrVelvetFog**
## Description
Hardens MLX safetensors loading in `openmed.mlx.models.load_model` and adds an `OPENMED_MLX_MMAP` toggle so callers can force eager loading for debugging. Closes #296.

**Implementation note (MLX API):** the issue lists "`mx.load` with mmap" as one option, but MLX 0.31's `mx.load(file, /, format=None, return_metadata=False, *, stream=None)` has no `mmap` parameter. MLX already reads `.safetensors` lazily and memory-mapped at the native level (tensors are not resident until evaluated), so this PR takes the issue's other stated option — lazy load — as the default "mmap path" and uses the toggle to force eager materialization as the documented fallback. The default loading behavior is therefore unchanged; the change adds the explicit toggle, documents the lazy/mmap behavior, and locks correctness in with tests. Happy to adjust if you'd prefer a different shape.

## Type of Change
- [x] Performance improvement
- [x] Test addition/improvement

## Changes Made
- `openmed/mlx/models/__init__.py`: route the safetensors/npz read through a new `_load_mlx_weights` helper. By default it uses MLX's lazy, memory-mapped path; when `OPENMED_MLX_MMAP` is `0`/`false`/`no`/`off` it eagerly evaluates every tensor up front as a documented fallback. Added `_mlx_mmap_enabled()` to parse the toggle (unset/empty → enabled).
- Quantized (`.scales`) and non-quantized checkpoints both load unchanged.
- `tests/unit/mlx/test_mmap_loading.py`: toggle parsing, eager/lazy `mx.eval` behavior (mocked, no MLX required), and a real-MLX safetensors round-trip with mmap enabled and disabled (skipped when MLX is unavailable).

## Testing
- [x] Added tests; full unit suite passes locally (`.venv/bin/python -m pytest tests/unit -q` → 2983 passed, 25 skipped), with MLX installed so the existing MLX loader tests exercise the change.

## Documentation
- [x] Added docstrings to the new helpers and updated `CHANGELOG.md`.

## Code Quality
- [x] Ran Ruff format and lint (`ruff format --check .` and `ruff check .` are clean repo-wide).
- [x] Performed a self-review.

## Dependencies
- [x] No new package dependencies.

## Related Issues
Closes #296

### CODE REVIEW

**Comment by maziyarpanahi** [comment]
Thank you @DrVelvetFog. I reviewed this against #296 and current `master`, resolved the stale `CHANGELOG.md` conflict, and merged it with commit `25f6a23d`.

What changed:
- Added the `OPENMED_MLX_MMAP` toggle for the MLX weight-loading path.
- Kept mmap/lazy loading enabled by default and added a documented eager fallback for debugging.
- Added mocked no-hardware tests plus optional real-MLX roundtrip coverage for enabled and disabled mmap modes.

Validation:
- `uv run --extra dev ruff format openmed/mlx/models/__init__.py tests/unit/mlx/test_mmap_loading.py`
- `uv run --extra dev ruff check openmed/mlx/models/__init__.py tests/unit/mlx/test_mmap_loading.py`
- `uv run --extra dev pytest tests/unit/mlx/test_mmap_loading.py -q` (16 passed, 2 skipped)

This satisfies the issue and has been merged.

**Comment by DrVelvetFog** [comment]
Thanks for the careful review and the conflict cleanup on both — much appreciated. Glad the mmap toggle landed with the honest framing intact. openmed's a great codebase to contribute to; happy to pick up another issue down the line if something fits.
